### PR TITLE
Appointment request search enhancement

### PR DESCRIPTION
**Pull request by sujanb12**
Description
----
Added a custom date search parameter '-date-or-req-period' for appointment search.
The previous PR was closed since changes were not deployed. https://github.com/cerner/fhir.cerner.com/pull/476

Screen Shot 2021-03-10 at 7 18 43 AM
Screen Shot 2021-03-10 at 7 23 03 AM
Screen Shot 2021-03-10 at 7 23 45 AM
Screen Shot 2021-03-10 at 7 24 33 AM

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

File: content/millennium/r4/scheduling/appointment.md [hunk 1]
```diff
@@ -66,23 +66,26 @@ _Implementation Notes_
...
+ [`_count`]            | No                                                                                 | [`number`]    | The maximum number of results to return.
 
 Notes:
 
 * The `patient`, `practitioner`, and `location` parameters may be included only once and may not be used in combination. For example, `patient=1234,5678` is supported but `patient=1234&patient=5678` and `patient=1234&location=5678` are not.
-* The `date` parameter may be provided:
+* Either `date` or `-date-or-req-period` parameter may be provided:
   * once with a prefix and time component to indicate a specific date/time. (e.g. `&date=ge2019-12-07T22:22:16.270Z`, `&date=lt2019-12-14T22:22:16.270Z`)
   * twice with the prefixes `ge` and `lt` to indicate a specific range. The date and prefix pairs must define an upper and lower bound. (e.g. `&date=ge2019-12-07T22:22:16.270Z&date=lt2019-12-14T22:22:16.270Z`)
+* The `_count` parameter, if provided must be a number not less than 5 and not greater than 100.
```
Comment (Coffeevee) [inline content/millennium/r4/scheduling/appointment.md:86]: nit: The `_count` parameter, if provided, must...

Comment (sujanb12) [reply content/millennium/r4/scheduling/appointment.md]: updated [here](https://github.com/cerner/fhir.cerner.com/pull/575/commits/ba209485b9351f67c53de40269561b09d4d47f17#diff-af2fad7ff605afe25e51152d541f652713df8ea51a020d1eae9bc557c158d2cbR86)

File: content/millennium/r4/scheduling/appointment.md [hunk 2]
```diff
@@ -66,23 +66,26 @@ _Implementation Notes_
...
+ [`_count`]            | No                                                                                 | [`number`]    | The maximum number of results to return.
 
 Notes:
 
 * The `patient`, `practitioner`, and `location` parameters may be included only once and may not be used in combination. For example, `patient=1234,5678` is supported but `patient=1234&patient=5678` and `patient=1234&location=5678` are not.
-* The `date` parameter may be provided:
+* Either `date` or `-date-or-req-period` parameter may be provided:
   * once with a prefix and time component to indicate a specific date/time. (e.g. `&date=ge2019-12-07T22:22:16.270Z`, `&date=lt2019-12-14T22:22:16.270Z`)
   * twice with the prefixes `ge` and `lt` to indicate a specific range. The date and prefix pairs must define an upper and lower bound. (e.g. `&date=ge2019-12-07T22:22:16.270Z&date=lt2019-12-14T22:22:16.270Z`)
+* The `_count` parameter, if provided, must be a number not less than 5 and not greater than 100.
```
Comment (alexbeckner1) [inline content/millennium/r4/scheduling/appointment.md:86]: Would it be better to rephrase this to "The `_count` parameter must be a number not less than 5 and not greater than 100 when provided."

Comment (sujanb12) [reply content/millennium/r4/scheduling/appointment.md]: updated [here](https://github.com/cerner/fhir.cerner.com/pull/575/commits/a7341d6cb04d70ee3d397d9aff55ab9b0f16da47#diff-af2fad7ff605afe25e51152d541f652713df8ea51a020d1eae9bc557c158d2cbR86)

Comment (alexbeckner1) [reply content/millennium/r4/scheduling/appointment.md]: Thanks!

**Review by alexbeckner1 — APPROVED** [review]
+1

File: content/millennium/r4/scheduling/appointment.md [hunk 3]
```diff
@@ -66,23 +66,26 @@ _Implementation Notes_
...
+ [`_count`]            | No                                                                                 | [`number`]    | The maximum number of results to return.
 
 Notes:
 
 * The `patient`, `practitioner`, and `location` parameters may be included only once and may not be used in combination. For example, `patient=1234,5678` is supported but `patient=1234&patient=5678` and `patient=1234&location=5678` are not.
-* The `date` parameter may be provided:
+* Either `date` or `-date-or-req-period` parameter may be provided:
   * once with a prefix and time component to indicate a specific date/time. (e.g. `&date=ge2019-12-07T22:22:16.270Z`, `&date=lt2019-12-14T22:22:16.270Z`)
   * twice with the prefixes `ge` and `lt` to indicate a specific range. The date and prefix pairs must define an upper and lower bound. (e.g. `&date=ge2019-12-07T22:22:16.270Z&date=lt2019-12-14T22:22:16.270Z`)
+* The `_count` parameter must be a number not less than 5 and not greater than 100 when provided.
```
Comment (exopsRamba) [inline content/millennium/r4/scheduling/appointment.md:86]: nit: it might be more readable to just say bust be a number greater than or equal to 5 and less than or equal to 100 when provided. Thoughts?

File: content/millennium/r4/scheduling/appointment.md [hunk 4]
```diff
@@ -66,23 +66,26 @@ _Implementation Notes_
...
+ `status`              | No                                                                                 | [`token`]     | A single or comma separated list of appointment statuses. Example: `arrived`
+ [`_count`]            | No                                                                                 | [`number`]    | The maximum number of results to return.
 
 Notes:
 
 * The `patient`, `practitioner`, and `location` parameters may be included only once and may not be used in combination. For example, `patient=1234,5678` is supported but `patient=1234&patient=5678` and `patient=1234&location=5678` are not.
-* The `date` parameter may be provided:
+* Either `date` or `-date-or-req-period` parameter may be provided:
   * once with a prefix and time component to indicate a specific date/time. (e.g. `&date=ge2019-12-07T22:22:16.270Z`, `&date=lt2019-12-14T22:22:16.270Z`)
   * twice with the prefixes `ge` and `lt` to indicate a specific range. The date and prefix pairs must define an upper and lower bound. (e.g. `&date=ge2019-12-07T22:22:16.270Z&date=lt2019-12-14T22:22:16.270Z`)
```
Comment (exopsRamba) [inline content/millennium/r4/scheduling/appointment.md:85]: Should we capitalize these? Might not matter since it already exists today.

File: content/millennium/r4/scheduling/appointment.md [hunk 5]
```diff
@@ -66,23 +66,26 @@ _Implementation Notes_
...
+* The `_count` parameter must be a number not less than 5 and not greater than 100 when provided.
 * The retrieved appointments are sorted first by `start` date ascending (earliest first), followed by the provided search parameter (`patient`, `practitioner` or `location`) and `start` time ascending (earliest first).
+* Search by `date` returns appointments that start on the date provided whereas search by `-date-or-req-period` returns appointments that start or are requested to start on the dates provided.
```
Comment (exopsRamba) [inline content/millennium/r4/scheduling/appointment.md:88]: nit: Comma before 'whereas'

Also doesn't -date-or-req-period match for appointments that start or are requested to start between the dates provided?

File: content/millennium/r4/scheduling/appointment.md (hunk 4 above)
Comment (sujanb12) [reply content/millennium/r4/scheduling/appointment.md:85]: I think these were not capitalized as they are a part of the previous statement. We can update it if we want but I think this is fine as well.

File: content/millennium/r4/scheduling/appointment.md (hunk 3 above)
Comment (sujanb12) [reply content/millennium/r4/scheduling/appointment.md]: Updated [here](https://github.com/cerner/fhir.cerner.com/pull/575/commits/f72b2fdc7548c22e35333e1403b64677ed9b5ca1#diff-af2fad7ff605afe25e51152d541f652713df8ea51a020d1eae9bc557c158d2cbR86)

**Comment by sujanb12** [comment]
Updated notes:

Screen Shot 2021-03-16 at 11 38 57 AM

File: content/millennium/r4/scheduling/appointment.md [hunk 6]
```diff
@@ -66,22 +66,26 @@ _Implementation Notes_
...
- [`_count`]     | No                                                       | [`number`]    | The maximum number of results to return.
+ Name                  | Required?                                                                          | Type          | Description
+-----------------------|------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------
+ `_id`                 | Yes, or one of `patient`, `practitioner`, or `location`.                           | [`token`]     | The logical resource id associated with the Appointment. Example: `3005759`
+ `date`                | Yes, or `-date-or-req-period` when using `patient`, `practitioner`, or `location`. | [`date`]      | The Appointment start date time with offset. Example: `2019-06-07T22:22:16.270Z`
+ `-date-or-req-period` | Yes, or `date` when using `patient`, `practitioner`, or `location`.                | [`date`]      | The Appointment start instant or the Appointment requested period date time with offset. Example: `2019-06-07T22:22:16.270Z`
```
Comment (exopsRamba) [inline content/millennium/r4/scheduling/appointment.md:73]: Let's make the first part of this line match date. i.e. 'The appointment start date with offset or the Appointment requested period date time with offset'

Comment (sujanb12) [reply content/millennium/r4/scheduling/appointment.md]: updated [here|https://github.com/cerner/fhir.cerner.com/pull/575/commits/e81f3335620756bd3017f934c96d201dd3f3332f#diff-af2fad7ff605afe25e51152d541f652713df8ea51a020d1eae9bc557c158d2cbR73], [Screenshot here](https://github.com/cerner/fhir.cerner.com/pull/575#issuecomment-800425819)

Comment (exopsRamba) [reply content/millennium/r4/scheduling/appointment.md]: nit: let's capitalize Appointment to stay consistent.

Comment (sujanb12) [reply content/millennium/r4/scheduling/appointment.md]: https://github.com/cerner/fhir.cerner.com/pull/575/commits/76edf9abc53fb2057551d9f3d0b930d6302d3bf7#diff-af2fad7ff605afe25e51152d541f652713df8ea51a020d1eae9bc557c158d2cbR73

**Review by exopsRamba — APPROVED** [review]

**Review by Coffeevee — APPROVED** [review]

File: content/millennium/r4/scheduling/appointment.md [hunk 7]
```diff
@@ -66,22 +66,26 @@ _Implementation Notes_
...
+ [`_count`]            | No                                                                                 | [`number`]    | The maximum number of results to return.
 
 Notes:
 
 * The `patient`, `practitioner`, and `location` parameters may be included only once and may not be used in combination. For example, `patient=1234,5678` is supported but `patient=1234&patient=5678` and `patient=1234&location=5678` are not.
-* The `date` parameter may be provided:
+* Either `date` or `-date-or-req-period` parameter may be provided:
   * once with a prefix and time component to indicate a specific date/time. (e.g. `&date=ge2019-12-07T22:22:16.270Z`, `&date=lt2019-12-14T22:22:16.270Z`)
   * twice with the prefixes `ge` and `lt` to indicate a specific range. The date and prefix pairs must define an upper and lower bound. (e.g. `&date=ge2019-12-07T22:22:16.270Z&date=lt2019-12-14T22:22:16.270Z`)
+* The `_count` parameter must be a number between 5 and 100 inclusive when provided.
```
Comment (MadhurRaj) [inline content/millennium/r4/scheduling/appointment.md:86]: we should move the _count point below the dates, so the notes for the dates are all in one place.

Comment (sujanb12) [reply content/millennium/r4/scheduling/appointment.md]: updated [here](https://github.com/cerner/fhir.cerner.com/pull/575/commits/ee36db0f14e29535ed124134a681b1b6110dfada#diff-af2fad7ff605afe25e51152d541f652713df8ea51a020d1eae9bc557c158d2cbR89)

**Review by MadhurRaj — APPROVED** [review]

**Comment by BS062721** [comment]
Verified on live
Screen Shot 2021-03-18 at 2 56 52 PM

Screen Shot 2021-03-18 at 2 56 08 PM
